Assets/World/Units/Ships/Merchants/Huker/Huker.tscn

```diff
@@ -0,0 +1,59 @@
+[gd_scene load_steps=8 format=2]
+
+[ext_resource path="res://Assets/World/Units/Ships/Ship.tscn" type="PackedScene" id=1]
+[ext_resource path="res://Assets/World/Units/Ships/Merchants/Huker/Huker.gd" type="Script" id=2]
+[ext_resource path="res://Assets/World/Units/Ships/Merchants/Huker/Sprites/Huker_idle.png" type="Texture" id=3]
+[ext_resource path="res://Assets/Player/FactionColor/FactionColorNone.tres" type="Material" id=4]
+
+[sub_resource type="CylinderMesh" id=1]
+
+[sub_resource type="SpatialMaterial" id=2]
+flags_transparent = true
+flags_no_depth_test = true
+flags_do_not_receive_shadows = true
+flags_disable_ambient_light = true
+albedo_color = Color( 1, 1, 1, 0.117647 )
+
+[sub_resource type="Animation" id=3]
+length = 3.0
+loop = true
+
+[node name="Huker" index="0" instance=ExtResource( 1 )]
+script = ExtResource( 2 )
+texture = ExtResource( 3 )
+
+[node name="SelectionRing" type="MeshInstance" parent="." index="0"]
+transform = Transform( 1, 0, 0, 0, 0.005, 0, 0, 0, 1, 0, 0, 0 )
+visible = false
+mesh = SubResource( 1 )
+material/0 = SubResource( 2 )
+
+[node name="Billboard" parent="." index="1"]
+extra_cull_margin = 50.0
+texture = ExtResource( 3 )
+
+[node name="CollisionShape" parent="Area" index="0"]
+visible = true
+
+[node name="StaticBody" parent="." index="3"]
+visible = false
+
+[node name="FactionColor" type="Sprite3D" parent="." index="4"]
+material_override = ExtResource( 4 )
```

Comment:
Currently the ship and faction color draw exactly on top of each other, so it's a 50/50 chance which one you will actually see. Adding this line to the FactionColor entry in the TSCN ensures the faction color is always on top:

    transform = Transform( 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, -0.0001, 0 )

It's possible there's a better fix, though, it might be better to do this in code when spawning ships and other units to ensure it's done everywhere.

Reply:
I've figured by now that using the color as texture mask in a shader within the billboard sprite should be the actual way to go (also performance-wise). I'm not too familiar with shaders though as of now, so this will be something to note for later (unless someone else wants to implement it). In the meantime, your suggestion will do the job.